### PR TITLE
Hunter commit

### PR DESCRIPTION
**Pull request by akalsi87**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+install(TARGETS uv
+  EXPORT libuvConfig
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib)
+
+install(DIRECTORY include/ DESTINATION include)
+
+install(EXPORT libuvConfig DESTINATION lib/cmake/libuv)
```
Comment (ruslo) [inline CMakeLists.txt:119]: It's not enough. We should have an extra file where you can check the correctness of the `find_package` call (list of component, which should be empty in our case) and the dependency (Threads in our case). See
* https://github.com/forexample/package-example/blob/cf2ea1d6a209fb9eca2ab83fdd0ac15fe4d3e807/Foo/CMakeLists.txt#L64-L147
* https://github.com/forexample/package-example/blob/cf2ea1d6a209fb9eca2ab83fdd0ac15fe4d3e807/Foo/cmake/Config.cmake.in

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,119 @@
+cmake_minimum_required(VERSION 2.8)
```
Comment (ruslo) [inline CMakeLists.txt:1]: `project` command should be right after `cmake_minimum_required`:
* http://cgold.readthedocs.io/en/latest/tutorials/project.html#when-not-declared

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+endif()
+
+add_library(uv ${SOURCES})
+
+find_package(Threads)
+
+if(WIN32)
+  target_link_libraries(uv ws2_32 psapi iphlpapi shell32 userenv)
+else()
+  target_link_libraries(uv ${CMAKE_THREAD_LIBS_INIT})
```
Comment (ruslo) [inline CMakeLists.txt:101]: it's better to use `target_link_libraries(uv PUBLIC Threads::Threads)` if you want package to be relocatable:
* https://cmake.org/cmake/help/latest/module/FindThreads.html

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    uv
+    PRIVATE BUILDING_UV_SHARED
+    INTERFACE USING_UV_SHARED)
+endif()
+
+install(TARGETS uv
+  EXPORT libuvConfig
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib)
```
Comment (ruslo) [inline CMakeLists.txt:115]: We need `INCLUDE DESTINATION` here:
* https://github.com/forexample/package-example/blob/cf2ea1d6a209fb9eca2ab83fdd0ac15fe4d3e807/Foo/CMakeLists.txt#L105-L112

Comment (akalsi87) [reply CMakeLists.txt:115]: Does this mean that the headers must be part of sources?

Comment (ruslo) [reply CMakeLists.txt:115]: No. Headers installed in line # 119. Here we just propagating them to the users. It's something like 
```cmake
target_include_directories(uv PUBLIC "${CMAKE_INSTALL_PREFIX}/include)")
```
 (but not exactly).

From [docs](https://cmake.org/cmake/help/latest/command/install.html):
```
The INCLUDES DESTINATION specifies a list of directories which will be added to
the INTERFACE_INCLUDE_DIRECTORIES target property of the <targets> when
exported by the install(EXPORT) command. If a relative path is specified, it is
treated as relative to the $<INSTALL_PREFIX>. This is independent of the rest
of the argument groups and does not actually install anything.
```

**Pull request merged**